File: adventofcode.com/2024/01/1.rb
list_1 = []
list_2 = []

File.open(File.dirname(__FILE__) + "/input.txt", "r") do |f|
  f.each_line do |line|
    values = line.strip.split('   ')
    list_1.push(Integer(values[0]))
    list_2.push(Integer(values[1]))
  end
end

list_1 = list_1.sort
list_2 = list_2.sort

distance = 0
(0..list_1.length - 1).each do |i|
  distance += (list_1[i] - list_2[i]).abs
end

puts(distance)  # 1580061

File: adventofcode.com/2024/01/1.py
import os

list_1 = []
list_2 = []

with open(os.path.dirname(__file__) + '/input.txt', mode='r') as f:
    for raw_line in f.readlines():
        values = raw_line.strip().split('   ')
        list_1.append(int(values[0]))
        list_2.append(int(values[1]))

list_1.sort()
list_2.sort()

distance = 0
for i in range(len(list_1)):
    distance += abs(list_1[i] - list_2[i])

print(distance)  # 1580061

